test/src/unit-cppapi-array.cc

```diff
@@ -1268,3 +1268,90 @@ TEST_CASE(
   if (vfs.is_dir(array_name))
     vfs.remove_dir(array_name);
 }
+
+TEST_CASE("C++ API: Test element offsets", "[cppapi][sparse][element-offset]") {
+  Context ctx;
+  VFS vfs(ctx);
+
+  // Create the array
+  std::string array_name = "test_element_offset";
+  if (vfs.is_dir(array_name))
+    vfs.remove_dir(array_name);
+
+  Domain dom(ctx);
+  dom.add_dimension(Dimension::create<int64_t>(ctx, "d1", {{1, 4}}, 4))
+      .add_dimension(Dimension::create<int64_t>(ctx, "d2", {{1, 4}}, 4));
+
+  ArraySchema schema(ctx, TILEDB_SPARSE);
+  Attribute attr(ctx, "attr", TILEDB_INT32);
+  attr.set_cell_val_num(TILEDB_VAR_NUM);
+  schema.add_attribute(attr);
+  schema.set_tile_order(TILEDB_ROW_MAJOR);
+  schema.set_cell_order(TILEDB_ROW_MAJOR);
+  schema.set_domain(dom);
+
+  Array::create(array_name, schema);
+
+  // Write some variable-sized data in the array
+  std::vector<int64_t> d1 = {1, 2, 3, 4};
+  std::vector<int64_t> d2 = {2, 1, 3, 4};
+
+  std::vector<int32_t> data = {1, 1, 2, 3, 3, 3, 4};
+  std::vector<uint64_t> data_offsets = {0, 8, 12, 24};
+
+  Array array(ctx, array_name, TILEDB_WRITE);
+  Query query(ctx, array, TILEDB_WRITE);
+  query.set_layout(TILEDB_UNORDERED)
+      .set_buffer("d1", d1)
+      .set_buffer("d2", d2)
+      .set_buffer("attr", data_offsets, data);
+  CHECK_NOTHROW(query.submit());
+
+  array.close();
+
+  // 1. READ when offset format is bytes (default case)
+  Config config = ctx.config();
+  CHECK((std::string)config["sm.offsets_format"] == "bytes");
```

Comment:
I don't think this needs an `std::string` cast.

Reply:
For some reason config["sm.offsets_format"] it's not automatically coverted from ConfigProxy to string, even if I compare it to std::string("bytes"), although it is supposed to. I get: `no operator "==" matches these operands -- operand types are: tiledb::impl::ConfigProxy == std::__1::string`

Any ideas why?

This cast is the same way the check is done in lines 95-96 of this same file:
` CHECK((std::string)cfg["vfs.s3.region"] == "us-east-1a");
  CHECK((std::string)cfg["vfs.s3.use_virtual_addressing"] == "true");
}`

Reply:
I see -- thanks. That's fine. Generally we prefer `static_cast<std::string>()` because it is slightly safer. I'll leave that up to you.